lib/is-ipv4-mapped-ipv6-address.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const { isIP } = require('node:net')
+
+/**
+ * Determine if `addr` is a IPv4-mapped IPv6 address.
+ *
+ * @param {string} addr
+ * @returns {boolean}
+ */
+function isIPv4MappedIPv6Address (addr) {
+  if (
+    addr[0] === ':' &&
+    addr[1] === ':'
+  ) {
+    if (
+      addr[2] === 'f' &&
```

Comment:
is symbol-by-symbol check faster than resolving slice and comparing it as a whole?

Reply:
from my experience: yes.